tests/communication/ros2/test_api.py

```
@@ -0,0 +1,511 @@
+def test_ros2_action_send_goal_wrong_action_type(
+    ros_setup: None, request: pytest.FixtureRequest
+) -> None:
+    action_name = f"{request.node.originalname}_action"  # type: ignore
+    node_name = f"{request.node.originalname}_node"  # type: ignore
+    action_server = ActionServer_(action_name)
+    node = Node(node_name)
+    executors, threads = single_threaded_spinner([action_server, node])
+
+    try:
+        action_api = ROS2ActionAPI(node)
+        with pytest.raises(Exception):
```

Ruff (0.8.2)

`pytest.raises(Exception)` should be considered evil

(B017)